### PR TITLE
Makefile cleanup

### PR DESCRIPTION
**Pull request by stephenfin**
Remove dead targets from `Makefile`

More information is provided in the commit messages.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -59,22 +51,22 @@ clean-buildfiles:
...
 	rm -rf .mypy_cache/
 
+.PHONY: style-check
+style-check:
+	@flake8
+
+.PHONY: type-check
+type-check:
+	mypy sphinx/
+
```
Comment (tk0miya) [inline Makefile:61]: off topic:  I'm using this on my local (especially type-check target). The `:make` command of vim can handle the result easier. So I'd like to keep them as is.
http://vim.wikia.com/wiki/Automatically_open_the_quickfix_window_on_:make

**Review by tk0miya — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (stephenfin) [reply Makefile:61]: Oooh, neat 😄 I generally have a second pane open in `tmux` and just jump between them. Maybe I should investigate `tox` integration for `vim` too

**EDIT:** no intention of removing these quite yet, don't worry

**Comment by tk0miya** [comment]
Merged!

**Pull request merged**